File: grunt-jsonconst/test/fixtures/photoalbum.ts
    /**
     * Photo album metadata:
     * Describes a photo album, and metadata about the images in it
     */
    export abstract class PhotoAlbum
    {
        /**
         * Album name:
         * The name of the photo album
         */
        public abstract get name(): string;
        /**
         * Photos in the album:
         * Describes each of the photos in the album
         */
        public abstract get images(): ImageMetaData[];
    }

    /**
     * 0 schema.:
     * An explanation about the purpose of this instance.
     */
    export abstract class ImageMetaData
    {
        /**
         * Date schema.:
         * An explanation about the purpose of this instance.
         */
        public abstract get date(): string;
        /**
         * Photographer schema.:
         * An explanation about the purpose of this instance.
         */
        public abstract get photographer(): string;
    }


    class PhotoAlbum1 extends PhotoAlbum
    {
        public get name(): string
        {
            return "Album 1";
        }
        public get images(): ImageMetaData[]
        {
            return [new ImageMetaData2(), ];
        }
    }

    class ImageMetaData2 extends ImageMetaData
    {
        public get date(): string
        {
            return "March 72";
        }
        public get photographer(): string
        {
            return "Thomas";
        }
    }


    export default new PhotoAlbum1();

File: grunt-jsonconst/test/fixtures/photoalbum.cs
namespace x
{
    /// <summary>
    /// Photo album metadata:
    /// Describes a photo album, and metadata about the images in it
    /// </summary>
    public interface PhotoAlbum
    {
        /// <summary>
        /// Album name:
        /// The name of the photo album
        /// </summary>
        string name { get; } 
        /// <summary>
        /// Photos in the album:
        /// Describes each of the photos in the album
        /// </summary>
        ImageMetaData[] images { get; } 
    }

    /// <summary>
    /// 0 schema.:
    /// An explanation about the purpose of this instance.
    /// </summary>
    public interface ImageMetaData
    {
        /// <summary>
        /// Date schema.:
        /// An explanation about the purpose of this instance.
        /// </summary>
        string date { get; } 
        /// <summary>
        /// Photographer schema.:
        /// An explanation about the purpose of this instance.
        /// </summary>
        string photographer { get; } 
    }


    class PhotoAlbum1: PhotoAlbum
    {
        public string name
        {
            get
            {
                return "Album 1";
            }
        }
        
        public ImageMetaData[] images
        {
            get
            {
                return new ImageMetaData[] { new ImageMetaData2(), };
            }
        }
        
    }

    class ImageMetaData2: ImageMetaData
    {
        public string date
        {
            get
            {
                return "March 72";
            }
        }
        
        public string photographer
        {
            get
            {
                return "Thomas";
            }
        }
        
    }


    const PhotoAlbum y = new PhotoAlbum1();
}
